tests/ft_elm.vader

```diff
@@ -0,0 +1,65 @@
+Include: include/setup.vader
+
+Execute (elmMake parse success message):
+  let json = 'Successfully generated /dev/null'
+
+  Save &errorformat
+  let &errorformat = neomake#makers#ft#elm#elmMake().errorformat
+  let error = neomake#makers#ft#elm#ElmMakeMapexpr(json)
+
+  AssertEqual 0, error
+
+Execute (elmMake error message):
+  let json = '[{"tag":"BAD MAIN TYPE","overview":"The `main` value has an unsupported type.","subregion":null,"details":"I need Html, Svg, or a Program so I have something to render on screen, but you\ngave me:\n\n    String","region":{"start":{"line":6,"column":1},"end":{"line":6,"column":5}},"type":"error","file":"Bingo.elm"}]'
+
+  Save &errorformat
+  let &errorformat = neomake#makers#ft#elm#elmMake().errorformat
+
+  let error = neomake#makers#ft#elm#ElmMakeMapexpr(json)
+
+  AssertEqual '[E] "Bingo.elm" 6:1 4 BAD MAIN TYPE : The `main` value has an unsupported type.', error
+
+  cgetexpr error
+
+  let result = getqflist()[0]
+  unlet result.bufnr
```

Comment:
Yeah, an alternative appears to be checking that `bufnr` equals `bufnr('%') + 1`, but that feels brittle. I'm not that familiar with vimscripting so any advice would be appreciated.